### PR TITLE
Add multi-flavor support

### PR DESCRIPTION
**Pull request by dkart**
You can issue:
```
$ grunt build:<flavor>
```
in order to build your flavor of preference, or leave it unspecified so that all available flavors are built in the respective subdirectories under `dist` directory.

### CODE REVIEW

File: Gruntfile.coffee [hunk 1]
```diff
@@ -13,11 +13,20 @@ module.exports = (grunt) ->
...
+
     clean:
       payload:
-        src: ['dist/js/payload*']
+        src: ["dist/<%= grunt.config.get('current_flavor') %>/js/payload*"]
       plugins:
-        src: ['dist/js/plugins/*']
+        src: ["dist/<%= grunt.config.get('current_flavor') %>/js/plugins/*"]
+      samples:
+        src: ['src/settings.coffee.sample.1', 'src/plugins_settings.coffee.sample.1']
```
Comment (kosdiamantis) [inline Gruntfile.coffee]: What's the purpose of this sample?

Comment (dkart) [inline Gruntfile.coffee]: I've added a preprocessing step for the samples, where `@@flavor` is replaced with the flavor being built. So:
```
@@@@flavor.analytics_base_url
```
becomes:
```
@@skroutz.analytics_base_url
```
The latter is finally replaced depending on the configuration in the `.yml` file:
```
skroutz:
  analytics_base_url: "https://analytics.skroutz.gr"
  application_base_url: "https://skroutz.gr"
```

File: Gruntfile.coffee [hunk 2]
```diff
@@ -279,6 +302,13 @@ module.exports = (grunt) ->
 
         command: 'rm -rf compiled dist src/{plugins_,}settings.coffee'
 
+      clean_dist:
+        options:
+          stdout: true
+
+        command: 'rm -rf compiled dist src/{plugins_,}settings.coffee'
```
Comment (kosdiamantis) [inline Gruntfile.coffee]: Why we need to duplicate this one? 
https://github.com/skroutz/analytics.js/pull/28/files#diff-a35864fcc4749a3b904641b4be9feacfL280

Comment (dkart) [inline Gruntfile.coffee]: We don't! It's just a piece of code I've accidentally left there 😄

File: Gruntfile.coffee [hunk 3]
```diff
@@ -35,27 +43,40 @@ module.exports = (grunt) ->
...
-        src: 'dist/js/payload.js'
-        dest: 'dist/js'
+        src: "dist/tmp/js/payload.js"
+        dest: "dist/tmp/js"
 
       plugins:
-        src: 'dist/js/plugins/*'
-        dest: 'dist/js/plugins/'
+        src: "dist/js/tmp/plugins/*"
+        dest: "dist/js/tmp/plugins/"
```
Comment (kosdiamantis) [inline Gruntfile.coffee]: [typo] 'dist/**tmp**/**js**/plugins'

File: Gruntfile.coffee [hunk 4]
```diff
@@ -132,6 +132,19 @@ module.exports = (grunt) ->
           dest: '.tmp/'
         ]
 
+      plugins_translations:
+        options:
+          patterns: [json: (done) ->
+            done(grunt.file.readYAML("locale/#{grunt.config.get('current_flavor')}/translations.yml"))
```
Comment (dkart) [inline Gruntfile.coffee]: I think that `locale` directory could be renamed to `flavor`.

File: Gruntfile.coffee [hunk 5]
```diff
@@ -116,6 +116,22 @@ module.exports = (grunt) ->
           dest: '.tmp/js/payload.js'
         ]
 
+      plugins_flavor:
+        options:
+          patterns: [
+            {
+              match: 'flavor'
+              replacement: -> grunt.config.get('current_flavor')
```
Comment (dkart) [inline Gruntfile.coffee]: This is equivalent to:
```
replacement: "<%= grunt.config.get('current_flavor') %>"
```

Templating has also been used [here](https://github.com/skroutz/analytics.js/blob/build_flavors/Gruntfile.coffee#L355); perhaps we should remain consistent and use a single approach.

**Comment by kosdiamantis** [comment]
👍 LGTM :shipit:

**Pull request merged**